### ISSUE TITLE
Toggle Switch knob - few-frames wrong rendering if starts with Checked

### ISSUE THREAD

**Issue by AndrejBunjac** [AvaloniaUI/Avalonia#7619]
**Describe the bug**
Toggle Switch that starts in a Checked position through a binding gets the know rendered as Unchecked for few frames when shown for the first time. This is visible in Control Catalog when choosing the ToggleSwitch tab the first time. In catalog it happens only once but in my project it happens on every load of the page because the switches are generated on loading so it looks really ugly.

**To Reproduce**
1. Run the ControlCatalog.
2. Select the ToggleSwitch tab for the first time.
3. Watch at the Headered Switch immediately after slecting.

**Expected behavior**
The know should be immediately rendered to the right if the toggle is checked. Instead it's rendered on the left for a few frames.

![Control Catalog toggle switch](https://user-images.githubusercontent.com/53874154/154242728-0db60250-5a2c-42d7-b6d8-74ae25313253.gif)

**Comment by maxkatz6** [comment]
IIRC, in case of toggle switch it is running because of value was changed. From initial false to true.
I had this workaround (basically replacing animation when it's not dragging or pointerover):
```
  <Style Selector="ToggleSwitch:not(:dragging):not(:pointerover) /template/ Grid#MovingKnobs">
    <Setter Property="Transitions">
      <Transitions>
        <DoubleTransition Property="Canvas.Left" Duration="0:0:0" Easing="CubicEaseOut"/>
      </Transitions>
    </Setter>
  </Style>
```

But better to investigate if there is better solution.

**Comment by timunie** [comment]
>  From initial false to true.

IMO if I set up the binding to true on the first run, no animation is expected. But I know that the system handles it that way. Was also an issue in WPF :-) 

> But better to investigate if there is better solution.

I agree. I think the animation should not run if the initial value is already true. But I didn't find a solution yet. I also think this would have performance benefit, when the animation does not run immidiatly. I think something needs to change when the control is attached to the visual tree. Also sometimes this may be intended, so maybe we need a special switch for this. ^^

**Comment by stogle** [comment]
I have a DataGrid with a column of ToggleSwitches where most are set to true and the issue is really obvious there. Due to virtualization, the controls get destroyed when they scroll out of view and recreated when you scroll back, so the transition happens over and over.

I also applied a style as a workaround to disable the transition, but I just used `<Setter Property="Transitions" Value="{x:Null}" />`. Note that in Avalonia 11.0 the Grid Name has changed from "MovingKnobs" to "PART_MovingKnobs".
